test/common/TestMnistCNN.hs

```diff
@@ -585,24 +587,11 @@ mnistCNNTestsLong = testGroup "MNIST CNN long tests"
               cx1 <- returnLet $ indexX cx ix1
               cx2 <- returnLet $ indexX cx1 ix2
               returnLet $ fromX0 cx2
-            ff = dFastForward f parameters ds
-            ffP@(_, ffPValue) = dFastForward fP parameters ds
-            perturbedffP@(_, perturbedffPValue) =
-              dFastForward fP parameters parametersPerturbation
-            close a b = abs (a - b) <= 1e-4
-            closeEq (a1, b1) (a2, b2) = close a1 a2 .&&. b1 === b2
-            dfDot fDot argsDot dsDot =
-              let (res, value) = dReverse 1 fDot argsDot
-              in (dotParameters res dsDot, value)
-        in ffPValue == perturbedffPValue
-           .&&. closeEq ff ffP
-           .&&. dForward f parameters ds === ff
-           .&&. closeEq (dfDot f parameters ds) ff
-           .&&. dForward fP parameters ds === ffP
-           .&&. closeEq (dfDot fP parameters ds) ffP
-           .&&. close (primalValue fP (addParameters parameters
-                                                     parametersPerturbation))
-                      (ffPValue + fst perturbedffP)
+        in
+            qcPropDom f  parameters ds parametersPerturbation 1 .&&.
+            qcPropDom fP parameters ds parametersPerturbation 1 .&&.
```

Comment:
A task for a separate PR (perhaps the one where we link the defining properties of gradients/derivatives to the Quickcheck properties in tests) is to also randomly generalte the `dt` values that arefixed at `1` here. They are fixed, because in an older version of the code, they were fixed at `1` somewhere deep in the internals of the library. But now they are exposed so we can vary them in tests.